android/app/src/main/java/au/com/robin/sms/service/SubscriberService.kt

```diff
@@ -49,6 +52,7 @@ class SubscriberService : Service() {
         super.onCreate()
```

Comment:
I wonder if `onStartCommand` and `onCreate` gets called at the same time.

Reply:
According to [this](https://developer.android.com/guide/components/services), `onStartCommand` gets called after `onCreate`. I quote it here,
> [onCreate()](https://developer.android.com/reference/android/app/Service#onCreate())
The system invokes this method to perform one-time setup procedures when the service is initially created (before it calls either [onStartCommand()](https://developer.android.com/reference/android/app/Service#onStartCommand(android.content.Intent,%20int,%20int)) or [onBind()](https://developer.android.com/reference/android/app/Service#onBind(android.content.Intent))). If the service is already running, this method is not called.

Reply:
does startForegrpund call onStartCommand?

Reply:
Yes, it does just just like `startService()`, the one provided by android, does. `onStartCommand` gets called each time we call `startService()`, the one in MainActivity with Intent. While `onCreate()` happens only once during the lifetime of service.